### DISCUSSION TITLE
q-file: Safari crashes when adding more than 1 file OR removing files with more than 1 file in queue

### DISCUSSION THREAD

**Discussion by madLinux7** [quasarframework/quasar#11071] [General - Components / Directives / etc]
```
<q-file append multiple ref="tileUploader" v-model="tileUploaderFiles" style="display: none"
                  @input="tileUploaderModelChange"
          />
```

I am using this invisible component to queue files for an upload mechanism. Everything works just fine in my browser, Android device or iOS simulator. But when im trying to add more files the the v-model after i already added one - or deleting one when more than 1 files are in queue - on my physical iPhone 8 (and the Safari Browser on Mac which i just found out) the app crashes.
Because capacitor apps run in an Safari web container, i just tested it in Safari on my mac. Here is the console log which is constantly spammed and causing the page to crash:

```
[Error] [Vue warn]: Duplicate keys detected: 'image.png'. This may cause an update error.

found in

---> <AuthenticatePage> at src/pages/MainPage.vue
       <QPageContainer>
         <QLayout>
           <MainLayout> at src/layouts/MenuLayout.vue
             <App> at src/App.vue
               <Root>
	warn (vue.runtime.esm.js:620)
	checkDuplicateKeys (vue.runtime.esm.js:6236)
	updateChildren (vue.runtime.esm.js:6174)
	patchVnode (vue.runtime.esm.js:6309)
	updateChildren (vue.runtime.esm.js:6183)
	patchVnode (vue.runtime.esm.js:6309)
	updateChildren (vue.runtime.esm.js:6183)
	patchVnode (vue.runtime.esm.js:6309)
	patch (vue.runtime.esm.js:6472)
	(anonyme Funktion) (vue.runtime.esm.js:3942)
	updateComponent (vue.runtime.esm.js:4060)
	get (vue.runtime.esm.js:4473)
	run (vue.runtime.esm.js:4548)
	flushSchedulerQueue (vue.runtime.esm.js:4304)
	(anonyme Funktion) (vue.runtime.esm.js:1979)
	flushCallbacks (vue.runtime.esm.js:1905)
	promiseReactionJob
```

I am not sure how to proceed now. What can be done to prevent Safari from creating duplicated keys? When I `console.log(this.tileUploaderFiles)` the output seems right. The file adding is handled by the q-file component. I handle the file removing by `this.tileUploaderFiles.splice(index, 1)`

In case the tileUploaderModelChange function is important (which i doubt), here it is (it handles displaying preview images in the tile view when a file is added) :

```
async tileUploaderModelChange () {
      console.log('FILES', this.tileUploaderFiles)
      if (this.tileUploaderFileAdded) { // depends on which operation got triggered
        let unspecifiedFileType = true
        const oldFileCount = this.tileUploaderFileCount
        const newFileCount = this.tileUploaderFiles.length
        const fileCountDifference = newFileCount - oldFileCount
        try {
          for (let i = 0; i < fileCountDifference; i++) {
            const base64 = await this.getBase64(this.tileUploaderFiles[oldFileCount + i])
            this.tileUploaderFileEditIndex++
            this.tileUploaderFileTitles.push(this.tileUploaderFiles[oldFileCount + i].name)
            if (base64.startsWith('data:image/jpeg') || base64.startsWith('data:image/png') || base64.startsWith('data:image/gif') ||
              base64.startsWith('data:image/svg')) {
              this.tileUploaderPreviews.push(base64)
              unspecifiedFileType = false
            }
            if (base64.startsWith('data:application/pdf')) {
              this.tileUploaderPreviews.push('/images/pdf-box.svg')
              unspecifiedFileType = false
            }
            if (base64.startsWith('data:application/vnd.oasis.opendocument') || base64.startsWith('data:application/vnd.openxmlformats-officedocument')) {
              this.tileUploaderPreviews.push('/images/file-table.svg')
              unspecifiedFileType = false
            }
            if (base64.startsWith('data:text/')) {
              this.tileUploaderPreviews.push('/images/text-box.svg')
              unspecifiedFileType = false
            }
            if (base64.startsWith('data:audio/')) {
              this.tileUploaderPreviews.push('/images/music-note.svg')
              unspecifiedFileType = false
            }
            if (base64.startsWith('data:video/')) {
              this.tileUploaderPreviews.push('/images/film.svg')
              unspecifiedFileType = false
            }
            // if no specified type
            if (unspecifiedFileType) {
              this.tileUploaderPreviews.push('/images/file.svg')
            }
          }
        } catch (e) {
          console.log('ERROR: ' + e.target.error.message)
          alert('ERROR: ' + e.target.error.message)
          location.reload()
        }
        this.tileUploaderFileCount = newFileCount
      }
      await this.CheckIfManualUploaderHasFiles()
    }
```
I hope someone has an idea for this problem. Thanks in advance!

(mockup look of the application)
![tileUploader](https://user-images.githubusercontent.com/50733707/138458725-00ed6140-8aa1-44c0-b73f-d741cfbe5e3b.png)

**Comment by hawkeye64** [comment]
@YourPenguin Based on the mimetype, you are displaying an image, correct? It also appears you are using this image as the key in a `v-for` loop.

```
[Error] [Vue warn]: Duplicate keys detected: 'image.png'. This may cause an update error.
```

In this case, it is `image.png`. Make sure all your keys are distinct. You could do this by also appending the index to the name.

```
:key="item.image + index"
```

**Reply by madLinux7** [reply]
nope, that's the point, i am not using a `v-for` loop, just the invisible (`style: "display: none"`) `q-file` component to store the file data in an automatic generated / expanded Array (I did not manually modified it at any point, everything is handled by q-file). I attached an image of the console log after inserting 3 files).
Almost every search regarding `[Vue warn]: Duplicate keys detected` results in a v-for context, which is pretty frustrating.

![Bildschirmfoto 2021-10-25 um 15 36 41](https://user-images.githubusercontent.com/50733707/138705738-d1259774-289c-4f2e-836c-1bbf5129df2d.png)

**Comment by hawkeye64** [comment]
In your other message, you said you were using v1.15.x -- did you upgrade to the latest v1.16.0? If so, are you experiencing the same issue?
